misc/python/materialize/mzbuild.py

```diff
@@ -250,18 +317,36 @@ def build(self) -> None:
         cargo_profile = "release" if self.rd.release_mode else "debug"
 
         def copy(exe: Path) -> None:
-            (self.path / exe).parent.mkdir(parents=True, exist_ok=True)
-            shutil.copy(
-                self.rd.cargo_target_dir() / cargo_profile / exe, self.path / exe
-            )
+            exe_path = self.path / exe
+            dbg_path = exe_path.with_suffix(exe_path.suffix + ".debug")
+            exe_path.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copy(self.rd.cargo_target_dir() / cargo_profile / exe, exe_path)
+
+            if self.split_debuginfo:
+                spawn.runv(
+                    [
+                        *self.rd.tool("objcopy"),
+                        exe_path,
+                        dbg_path,
+                        "--only-keep-debug",
+                    ],
+                    cwd=self.rd.root,
+                )
+                spawn.runv(
+                    [
+                        *self.rd.tool("objcopy"),
+                        exe_path,
+                        f"--add-gnu-debuglink={dbg_path}",
+                    ]
+                )
```

Comment:
I'm confused about this! Who is the debuglink for? It won't be useful when the binary is packaged in the Docker image because that Docker image won't have the .debug file, right?

Reply:
Discussed offline. We think this is only useful under very niche circumstances, so I will remove it after testing to see that the normal debugging flow still works without it.